torch/csrc/jit/codegen/cuda/executor_utils.cpp

```diff
@@ -1142,75 +1144,98 @@ std::pair<NvrtcFunction, std::string> nvrtcCompile(
-    }
+      std::vector<char> log(logsize);
+      at::globalContext().getNVRTC().nvrtcGetProgramLog(program, log.data());
 
-    ptxas_log << log.data() << std::endl;
-    if (isDebugDumpEnabled(DebugDumpOption::PrintPtxasLog)) {
-      std::cout << log.data() << std::endl;
+      if (result != NVRTC_SUCCESS) {
+        TORCH_INTERNAL_ASSERT(
+            false, code.c_str(), "\nCUDA NVRTC compile error: ", log.data());
+      }
+
+      ptxas_log << log.data() << std::endl;
+      if (isDebugDumpEnabled(DebugDumpOption::PrintPtxasLog)) {
+        std::cout << log.data() << std::endl;
+      }
+      AT_CUDA_NVRTC_CHECK(result);
     }
-    AT_CUDA_NVRTC_CHECK(result);
-  }
 
-  const char* lowered_kernel_name = nullptr;
-  at::globalContext().getNVRTC().nvrtcGetLoweredName(
-      program, func_name.c_str(), &lowered_kernel_name);
+    const char* lowered_kernel_name = nullptr;
+    at::globalContext().getNVRTC().nvrtcGetLoweredName(
+        program, func_name.c_str(), &lowered_kernel_name);
+    lowered_kernel_name_str.assign(lowered_kernel_name);
 
-  size_t ptx_size = 0;
-  // NOLINTNEXTLINE(cppcoreguidelines-init-variables)
-  std::vector<char> ptx;
+    size_t ptx_size = 0;
 
-  {
-    FUSER_PERF_SCOPE("executor_utils::Nvrtc::GetPTX");
+    {
+      FUSER_PERF_SCOPE("executor_utils::Nvrtc::GetPTX");
 #if CUDA_VERSION >= 11010
-    // compile_to_sass determines whether we are generating SASS or PTX, hence
-    // the different API.
-    const auto getSize = compile_to_sass
-        ? at::globalContext().getNVRTC().nvrtcGetCUBINSize
-        : at::globalContext().getNVRTC().nvrtcGetPTXSize;
-    const auto getFunc = compile_to_sass
-        ? at::globalContext().getNVRTC().nvrtcGetCUBIN
-        : at::globalContext().getNVRTC().nvrtcGetPTX;
+      // compile_to_sass determines whether we are generating SASS or PTX, hence
+      // the different API.
+      const auto getSize = compile_to_sass
+          ? at::globalContext().getNVRTC().nvrtcGetCUBINSize
+          : at::globalContext().getNVRTC().nvrtcGetPTXSize;
+      const auto getFunc = compile_to_sass
+          ? at::globalContext().getNVRTC().nvrtcGetCUBIN
+          : at::globalContext().getNVRTC().nvrtcGetPTX;
 #else
-    const auto getSize = at::globalContext().getNVRTC().nvrtcGetPTXSize;
-    const auto getFunc = at::globalContext().getNVRTC().nvrtcGetPTX;
+      const auto getSize = at::globalContext().getNVRTC().nvrtcGetPTXSize;
+      const auto getFunc = at::globalContext().getNVRTC().nvrtcGetPTX;
 #endif
-    AT_CUDA_NVRTC_CHECK(getSize(program, &ptx_size));
-    ptx.resize(ptx_size);
-    AT_CUDA_NVRTC_CHECK(getFunc(program, ptx.data()));
-  }
+      AT_CUDA_NVRTC_CHECK(getSize(program, &ptx_size));
+      ptx.resize(ptx_size);
+      AT_CUDA_NVRTC_CHECK(getFunc(program, ptx.data()));
+    }
 
-  NvrtcFunction compiled_kernel_;
+    if (kernel_db.enabled() && kernel_code.has_value()) {
+      kernel_db.write(
```

Comment:
nitpick: should throw some warnings when write fails. (warn_once maybe?)

Reply:
I added a warning.